libr/asm/arch/avr/assemble.c

```diff
@@ -366,3 +355,40 @@ int getnum(RAsm *a, const char *s) {
 	return r_num_math (a->num, s);
 }
 
+int assemble_general_io_operand(const char *operand, uint32_t *res) {
+	/*@
+		 requires strcmp (operand, "spl") == 0 || strcmp (operand, "0x3d") == 0 ||
+		 strcmp (operand, "sph") == 0 || strcmp (operand, "0x3e") == 0 ||
+		 strcmp (operand, "sreg") == 0 || strcmp (operand, "0x3f") == 0
+
+		 behavior case_spl
+		     assumes strcmp (operand, "spl") == 0 || strcmp (operand, "0x3d") == 0
+		     ensure *res == 0x3d
+		 behavior case_sph
+		     assumes strcmp (operand, "sph") == 0 || strcmp (operand, "0x3e") == 0
+		     ensure *res == 0x3d
+		 behavior case_sreg
+		     assumes strcmp (operand, "sreg") == 0 || strcmp (operand, "0x3f") == 0
+		     ensure *res == 0x3d
+
+		 complete behaviors;
+	     disjoint behaviors;
+	*/
+	int ret = -1;
+	if (strcmp (operand, "spl") == 0 || strcmp (operand, "0x3d") == 0) {
+		*res = 0x3d;
+		ret = 0;
+	} else if (strcmp (operand, "sph") == 0 || strcmp (operand, "0x3e") == 0) {
+		*res = 0x3e;
+		ret = 0;
+	} else if(strcmp (operand, "sreg") == 0 || strcmp (operand, "0x3f") == 0) {
+		*res = 0x3f;
+		ret = 0;
+	}/* else {
```

Comment:
Keep the else

Reply:
the else was just in case I forgot a possibility. I want to handle everything in order to avoid bug.

I also would like to remove it to make frama-c work. We could debate of frama-c here https://github.com/radareorg/radare2/issues/19469 if you want.